lib/middleware.js

```diff
@@ -320,7 +320,7 @@ class Kevin {
         // a function.
         const middleware = function kevinMiddleware(req, res, next) {
             // This is the name of the asset requested
-            const reqPath = req.path;
+            const reqPath = res.locals.overwrittenReqPath || req.path;
```

Comment:
Joe and I talked a bit about this — it seems like this change might be overly specific to Etsy's use of Kevin, and we want to make sure to keep Kevin flexible enough for others to use (we hope >_>). 

On like 351 below, we call `[getAssetName](https://github.com/etsy/kevin-middleware/blob/master/lib/middleware.js#L49)` which turns the request path into the location of an asset on disk. It currently only accepts `reqPath` as an argument. What if we extended it to accept `reqPath, req, res` as arguments? That would give you the ability to access `res.locals` [here,](https://github.etsycorp.com/Performance/buildapack/blob/535e4f4f0b69808b1cc917384b1d98a0b5764c7f/server.js#L35) in buildapack.

Reply:
That makes sense to me, and happy to update it to be more generic!

I noticed that reqPath gets used in Kevin in a few other spots, so I was nervous it would break these lines https://github.com/etsy/kevin-middleware/blob/master/lib/middleware.js#L328 if the reqPath is not actually find-able,  but I think it would be fine because these are specific checks for `__kevin` prefix in the reqPath, which wouldn't be affected by my change in Etsyweb to insert fake version strings. Just typing this out as a sanity check 😅 

I can update this PR to extend `getAssetName` to take in `req` and `res` params as well as `reqPath`. If I keep `reqPath`, then it shouldn't be a breaking change for Kevin, is that right?

Reply:
I think those other `reqPath` references are OK, since the change we're after is about getting the asset name rather than the request path.

Yeah, if you add `req, res` to `getAssetName`, it won't be a breaking change, which is nice!